client/requestUtil.js

```diff
@@ -420,6 +429,45 @@ RequestUtil.prototype.put = function (category, record) {
   })
 }
 
+/**
+ * Compact all records in a category
+ * @param {string=} category - the category ID
+ */
+RequestUtil.prototype.compactCategory = function (category) {
+  const thisCategory = category
+  if (!recordUtil.CATEGORY_IDS.includes(thisCategory)) {
+    throw new Error(`Unsupported sync category: ${category}`)
+  }
+
+  const prefix = `${this.apiVersion}/${this.userId}/${category}`
+  let options = {
+    MaxKeys: 1000,
+    Bucket: this.bucket,
+    Prefix: prefix
+  }
+  this.withRetry(() => {
+    return s3Helper.listObjects(this.s3, options, false)
+  }).then((s3Objects) => {
+    let latestRecords = {}
```

Comment:
this is going to keep the most recent record in each batch, which is fine, but I think we should add a comment to make that clear